File: PDOO/p4/Civitas/src/civitas/Casilla.java

package civitas;

import java.util.ArrayList;

public class Casilla {
    private String nombre;
    
    Casilla(String nombre){
        //Para crear la casilla de tipo descanso
        this.nombre = nombre;
    }
    
    public String getNombre(){
        return nombre;
    }
    
    void informe(int iactual, ArrayList<Jugador> todos){
        Diario.getInstance().ocurreEvento("El jugador " + todos.get(iactual).getNombre() + " ha caido en la casilla con informacion " + toString());
    }
    
    public Boolean jugadorCorrecto(int iactual, ArrayList<Jugador> todos){
        Boolean resultado= false;
        if (iactual >= 0 && iactual < todos.size()){
            resultado = true;
        }
        return resultado;
    }
    
    public void recibeJugador(int iactual, ArrayList<Jugador> todos){
       informe(iactual, todos);
    }
    
    @Override
    public String toString(){
        return "Casilla{ nombre=" + nombre + " }";
    }
            
}

File: PDOO/p4/civitas_rb/lib/casilla.rb
module Civitas

class Casilla
  
  def initialize(nombre)
    @nombre=nombre
  end
  
  def informe(iactual, todos)
    Diario.instance.ocurre_evento("El jugador #{todos.at(iactual).nombre} ha caido en la casilla con informacion #{to_string}")
  end
  
  def jugador_correcto(iactual, todos)
    resultado=false
    if ((iactual>=0) and (iactual <todos.length()))
      resultado=true
    end
    return resultado
  end
  
  def recibejugador(iactual, todos)
    informe(iactual, todos)
  end
  
  def to_string
    return "Casilla{ nombre= #{@nombre} }"
  end
  
  
  attr_accessor :nombre
end
end